packages/belos/tpetra/src/BelosTpetraOperator.hpp

```diff
@@ -0,0 +1,200 @@
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+
+/*! \file BelosTpetraOperator.h
+    \brief This file provides a Tpetra::Operator interface so Belos can be integrated into
+     other codes as an abstract operator.
+*/
+
+#ifndef BELOS_TPETRA_OPERATOR_H
+#define BELOS_TPETRA_OPERATOR_H
+
+#include "Tpetra_MultiVector_decl.hpp"
+#include "Tpetra_Operator.hpp"
+#include "Tpetra_Map.hpp"
+
+#include "BelosLinearProblem.hpp"
+#include "BelosStatusTest.hpp"
+#include "BelosOutputManager.hpp"
+#include "BelosTpetraAdapter.hpp"
+#include "BelosSolverFactory.hpp"
+
+#include "Teuchos_ParameterList.hpp"
+
+/*! \class Belos::TpetraOperator
+    \brief This class provides an interface to the Tpetra::Operator class, so Belos can be 
+    integrated into other codes as an abstract operator.
+*/
+
+namespace Belos {
+
+///////////////////////////////////////////////////////////////
+//-------- class BelosTpetraOperator --------------------
+//
+// This class will allow a Belos solver to be called as an Tpetra_Operator.
+// Thus, it can use itself as a preconditioner if need be.  It can
+// also be used as the inner iteration of Anasazi :)
+//
+///////////////////////////////////////////////////////////////
+template<class Scalar = ::Tpetra::Details::DefaultTypes::scalar_type,
```

Comment:
You can actually get the default types from `Tpetra::Operator`.  That way, you won't have to depend on an implementation detail of Tpetra.
```cpp
template<class Scalar = Tpetra::Operator<>::scalar_type,
  class LocalOrdinal = Tpetra::Operator<>::local_ordinal_type,
  class GlobalOrdinal = Tpetra::Operator<>::global_ordinal_type,
  class Node = Tpetra::Operator<>::node_type>
```